docs/building-apps/example-application-tutorial/querying-data.mdx

```diff
@@ -0,0 +1,335 @@
+        // @ts-ignore
+        throw error(err.response?.status ?? 400, {
+          // @ts-ignore
+          message: `${err.response?.title} - ${err.response?.detail}`,
+        });
+      }
+    }
+  } else {
+    throw error(400, { message: "invalid public key" });
+  }
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## fetchAccountBalances
+
+Gets existing balances for a given `publicKey`.
+
+```js title="/src/lib/stellar/horizonQueries.js"
+/**
+ * Fetches and returns balance details for an account on the Stellar network.
+ * @async
+ * @function fetchAccountBalances
+ * @param {string} publicKey Public Stellar address holding balances to query
+ * @returns {Promise<BalanceLine[]>} Array containing balance information for each asset the account holds
+ */
+export async function fetchAccountBalances(publicKey) {
+  const { balances } = await fetchAccount(publicKey);
+  return balances;
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## fetchRecentPayments
+
+Finds any payments made to or from the given `publicKey` (includes: payments, path payments, and account merges).
+
+```js title="/src/lib/stellar/horizonQueries.js"
+/**
+ * Fetches and returns recent `payment`, `createAccount` operations that had an effect on this account.
+ * @async
+ * @function fetchRecentPayments
+ * @param {string} publicKey Public Stellar address to query recent payment operations to/from
+ * @param {number} [limit] Number of operations to request from the server
+ * @returns {Promise<PaymentOperationRecord[]>} Array containing details for each recent payment
+ */
+export async function fetchRecentPayments(publicKey, limit = 10) {
+  const { records } = await server
+    .payments()
+    .forAccount(publicKey)
+    .limit(limit)
+    .order("desc")
+    .call();
+  return records;
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## submit
+
+Submit a signed transaction to the Stellar network.
+
+```js title="/src/lib/stellar/horizonQueries.js"
+/**
+ * Submits a Stellar transaction to the network for inclusion in the ledger.
+ * @async
+ * @function submit
+ * @param {Transaction} transaction Built transaction to submit to the network
+ * @throws Will throw an error if the transaction is not submitted successfully.
+ */
+export async function submit(transaction) {
+  try {
+    await server.submitTransaction(transaction);
+  } catch (err) {
+    throw error(400, {
+      // @ts-ignore
+      message: `${err.response?.title} - ${err.response?.data.extras.result_codes}`,
+    });
+  }
+}
+```
+
+**Source:** https://github.com/stellar/basic-payment-app/blob/main/src/lib/stellar/horizonQueries.js
+
+## fetchAssetsWithHomeDomains
+
+We create our own new types. Then, looks at all the issuer accounts, and returns only the ones with a `home_domain` set on the account.
```

Comment:
Need to tweak this wording, I wasn't quite sure what this meant.

Reply:
Thanks! I've tweaked it and added some more explanation, but you may want to give it a second glance to see if it needs anything else.